readme.md

```diff
@@ -691,7 +697,85 @@ coordsToPair(pairToCoords([1, 2])) // [1, 2]
 pairToCoords(coordsToPair({x: 1, y: 2})) // {x: 1, y: 2}
 ```
 
+### Homomorphism
 
+A homomorphism is just a structure preserving map. In fact, a functor is just a homomorphism between categories as it preserves the original category's structure under the mapping.
+
+```js
+A.of(f).ap(A.of(x)) == A.of(f(x))
+
+Either.of(_.toUpper).ap(Either.of("oreos")) == Either.of(_.toUpper("oreos"))
+```
+
+### Catamorphism
+
+A `reduceRight` function that applies a function against an accumulator and each value of the array (from right-to-left) to reduce it to a single value.
```

Comment:
I like this idea. Catamorphisms are more general than just a reduce and don't always require an accumulator